torch/fx/graph.py

```diff
@@ -64,10 +65,46 @@ def map_arg(a: Argument, fn: Callable[[Node], Argument]) -> Argument:
     else:
         return a
 
+class FinalizableList(list):
```

Comment:
Not sure all this infrastructure is worth it. We can just change `nodes` to be a property and then editing nodes doesn't change the graph.